### ISSUE TITLE
[BUG] `materialize_features` should forbid non-agg features

### ISSUE THREAD

**Issue by windoze** [feathr-ai/feathr#661]
### Willingness to contribute

Yes. I can contribute a fix for this bug independently.

### Feathr version

0.7.2

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 20.0)**:
- **Python version**:
- **Spark version, if reporting runtime issue**:

### Describe the problem

`materialize_features` shouldn't be used with non-agg features, as FeathrGenJob assumes the online storage to be a KV storage while the key is the agg key, it doesn't make sense to materialize non-agg features.

### Code to reproduce bug

```
client.materialize_features(['any_non_agg_feature'], materialize_settings)
```

### What component(s) does this bug affect?

- [X] `Python Client`: This is the client users use to interact with most of our API. Mostly written in Python.
- [X] `Computation Engine`: The computation engine that execute the actual feature join and generation work. Mostly in Scala and Spark.
- [ ] `Feature Registry API`: The frontend API layer supports SQL, Purview(Atlas) as storage. The API layer is in Python(FAST API)
- [ ] `Feature Registry Web UI`: The Web UI for feature registry. Written in React

**Comment by Yuqing-cat** [comment]
Issue Status Track: Pending on offline discussion (@windoze , @xiaoyongzhu ) about the correct way to fix it. 
But our example (recommendation) contains sample of non-agg features which will lead to misunderstanding. (currently provide #latest)

**Comment by blrchen** [comment]
This has been resolved by PR #719, closing
